.github/workflows/ci.yml

```
@@ -1,23 +1,40 @@
 name: C/C++ CI
 
-on: [pull_request]
+on: [push, pull_request]
 
 jobs:
-  CI:
-    # This job can be run locally through ./CI/build_test.sh
+  CI_host:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - name: Get submodules
-      run: git submodule update --init --checkout --recursive
-    - name: Build and test
-      run: |
-        ./CI/build_test.sh
+    - uses: actions/checkout@v2
+      with: 
+        submodules: true
+    - name: Build
+      run: docker-compose run --rm host ./CI/build_smooth_host.sh
+    - name: Test
+      run: docker-compose run --rm -w /src/build_host/test/linux_unit_tests host ./linux_unit_tests
+  
+  CI_esp:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        esp_idf_version:
```

Just to keep variable names different for less confusion.
```suggestion
        test_against_idf_versions:
```